### ISSUE TITLE
ION won't allow `accept` value for service objects

### ISSUE THREAD

**Issue by brianorwhatever** [decentralized-identity/didcomm-messaging#294]
I don't use ION but it looks like there might be some limitations here based on this discussion.

https://github.com/decentralized-identity/ion-tools/issues/16#issuecomment-925402311

I'm concerned other did methods will also have this limitation

**Comment by kdenhartog** [comment]
It's likely that this will be a problem depending on implementation design. The best thing we can do is to stabilize that service, register it, and then make sure that as many DID methods as we know are getting used can support it. I'm not sure that there's anything else we can do other than that.

**Comment by csuwildcat** [comment]
Folks should be putting random keys inside a `serviceEndpoint` object value to avoid polluting the top-level Service entry with foreign keys that could conflict with later additions to the DID Core spec. For example, I'd advise doing this:

```javascript
{
   id: '...',
   type: '...'
   serviceEndpoint: {
     routingKeys: ...
   } 
 } 
```

**Comment by brianorwhatever** [comment]
From the did core spec

> Each service extension MAY include additional properties and MAY further restrict the properties associated with the extension.

**Comment by csuwildcat** [comment]
> From the did core spec
> 
> > Each service extension MAY include additional properties and MAY further restrict the properties associated with the extension.

I'm not saying that route isn't allowed, just that I'd advise the safer route of constraining more elaborate configurations to completely cordoned off userland serviceEndpoint object values, as other Service types have.

**Comment by brianorwhatever** [comment]
can you link to the other service types?

**Comment by brianorwhatever** [comment]
The only examples in the did spec registry is this didcomm usecase, a few simple strings and an example of an array of strings with an `origins` key which to me is explaining the service endpoints, not how they work.

I am interested in seeing other folks who are using a complex object as a `serviceEndpoint` as to me this will involve more complicated handling than extending the top level service object node. I believe an endpoint should be an endpoint. Additional details about how that endpoint is used should be siblings not children.

**Comment by swcurran** [comment]
I don't understand @csuwildcat comment above with the `I'd advise doing this...` as what he proposes is (from what I can see) exactly what is in the DIDComm Service endpoint documented [here (Section 9.5 of the spec](https://identity.foundation/didcomm-messaging/spec/#did-document-service-endpoint).  What am I missing?

**Comment by swcurran** [comment]
Nevermind...I see the difference.  Comment deleted.

**Comment by TelegramSam** [comment]
Need to find a solution that works for ION. specify the structure of the object serviceendpoint type?

**Comment by TelegramSam** [comment]
Discussed WG 20220131. Need explicit examples of existing format for service endpoint, and explicit examples of an ION friendly way. Survey community to gather feedback about conflicts.

**Comment by TelegramSam** [comment]
Fixed with #352
